### PR TITLE
Go: Thrift support for tchannel (#318)

### PR DESCRIPTION
**Pull request by prashantv**
The client is compatible with node servers, as it skips writing the Thrift RPC call headers.

### CODE REVIEW

File: golang/thrift/output_protocol.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	// ReadArg2 has to be called before checking Response.ApplicationError.
+	reader, err := resp.Arg2Reader()
+	if err != nil {
+		return "", 0, 0, err
+	}
+	if err := reader.Close(); err != nil {
+		return "", 0, 0, err
+	}
+
+	if resp.ApplicationError() {
```
Comment (mmihic) [inline golang/thrift/output_protocol.go]: Don't you still need to unmarshal arg3 to extract the thrift exception body?

File: golang/thrift/output_protocol.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+// TChanOutboundOptions are parameters passed to the underlying tchannel when making requests.
+type TChanOutboundOptions struct {
+	// Context is the context.Context with timeout and deadline information.
+	Context context.Context
+
+	// Dst is the host:port of the service to call.
+	Dst string
+
+	// AutobahnService is the Autobahn service name.
```
Comment (mmihic) [inline golang/thrift/output_protocol.go]: nit: Hyperbahn

File: golang/thrift/output_protocol.go (hunk 1 above)
Comment (prashantv) [inline golang/thrift/output_protocol.go]: Yep, forgot to fix this, thanks

File: golang/thrift/output_protocol.go (hunk 2 above)
Comment (prashantv) [inline golang/thrift/output_protocol.go]: Done

**Comment by prashantv** [comment]
Travis is failing since thrift isn't available on it, we'll probably need to check in the thrift generated code. @mmihic can you think of any other alternatives?

**Comment by prashantv** [comment]
+ @weikai77 for comments, this is partially based on your PR.

**Pull request merged**